components/inspectit-ocelot-configurationserver/src/main/java/rocks/inspectit/ocelot/file/versioning/VersioningManager.java

```diff
@@ -444,7 +444,7 @@ WorkspaceDiff getWorkspaceDiff(boolean includeFileContent, ObjectId oldCommit, O
         List<SimpleDiffEntry> simpleDiffEntries = diffEntries.stream()
                 .map(SimpleDiffEntry::of)
                 .filter(entry -> entry.getFile().startsWith(AbstractFileAccessor.CONFIGURATION_FILES_SUBFOLDER))
-                .peek(entry -> {
+                .forEach(entry -> {
```

Comment:
This would not work because `forEach` terminates the stream, thus, the elements cannot be collected in a list.
If we want to replace `peek` we might have to use `map` and return the `entry` object after it has been modified. Maybe this would be the better way and by doing this, we'll align to the Javadocs and don't mutate the elements using `peek`.

Reply:
@mariusoe I have changed to `map` . Please check whether SimpleDiffEntry is the right place for the util method.